### ISSUE TITLE
Rendering right to left text does not work properly

### ISSUE THREAD

**Issue by ayousuf23** [MonoGame/MonoGame#7510]
When rendering a spritefont for a right-to-left language, the text appears in the wrong direction (left-to-right).

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- DesktopGL

**Comment by mrhelmut** [comment]
```SpriteFont``` currently doesn't support RTL rendering. It shouldn't be complicated to support it. If anyone wants to tackle this, I'm marking it as good first issue.

**Comment by ayousuf23** [comment]
What exactly would need to be modified to support RTL rendering?

**Comment by willmotil** [comment]
I imagine just a new overload of drawstring were you decrement instead of increment provided the data in spritefont has sufficient information.

**Comment by mrhelmut** [comment]
I would also recommend to make the starting cursor position to be on the right side of the first character, so that it's easier to align texts.

**Comment by mrhelmut** [comment]
If anyone is willing to give #7644 a test spin, that would be great. I would like to avoid doing that myself because I lack the cultural knowledge to make sure that it does things right.

**Comment by BlueElectivire** [comment]
I might try that tomorrow if I could find the time (and I manage to figure out how to do that).

**Comment by BlueElectivire** [comment]
So I tried to do that and I really didn't manage to test it, I couldn't figure out how to reference my working repo (Even when I removed the nuget references) and I'm not really familiar with writing in-code tests (especially not when it comes to graphics and stuff like that). If anyone is willing to try and explain it to me it'd be great.
Or if anyone would like to try that themselves and needs a quick explanation on how rtl text should look (Hebrew in my case) then I can gladly do that as well.

**Comment by mrhelmut** [comment]
@BlueElectivire Ah, by "testing" I meant trying to display text with it to verify that it renders as expected for people native to a RTL language. I didn't mean to make a unit code testing. Sorry for the confusion.

Let me know how it goes and I'll be happy to merge it if it's fine.

**Comment by BlueElectivire** [comment]
@mrhelmut so I tested it, turns out it didn't work.
I fixed it and committed the changes to the same branch, and now it seems to work fine.
I tested it with SpriteEffects applied as well and it works as I would expect it to.
The only thing is that SpriteFonts by default include only English and punctuation characters, and when you try to run it without changing the Spritefont file to include the necessary characters the game crashes but the error isn't visible anywhere so it was kinda hard to find (But I suppose that this would happen with other character sets that are left-to-right as well so this doesn't have anything to do with this really).

tl;dr: It didn't work but I fixed it and checked and it works now so I think you can merge it.
